.travis.yml

```diff
@@ -35,21 +31,15 @@ env:
 matrix:
   include:
     - python: 3.7
-      env: CACHE_NAME=C1.2 PANTS_SHA="1.25.x-twtr" IJ_ULTIMATE=false PANTS_TEST_JUNIT_TEST_SHARD=0/2
+      env: CACHE_NAME=C1.1 PANTS_SHA="1.25.x-twtr" IJ_ULTIMATE=false PANTS_TEST_JUNIT_TEST_SHARD=0/2
 
     - python: 3.7
-      env: CACHE_NAME=C2.2 PANTS_SHA="1.25.x-twtr" IJ_ULTIMATE=false PANTS_TEST_JUNIT_TEST_SHARD=1/2
-
-    - python: 3.7
-      env: CACHE_NAME=C3.2 PANTS_SHA="release_1.18.0" TEST_SET=jvm-integration
-
-    - python: 3.7
-      env: CACHE_NAME=C4.2 PANTS_SHA="release_1.16.0" TEST_SET=jvm-integration
+      env: CACHE_NAME=C2.2 PANTS_SHA="1.25.x-twtr" IJ_ULTIMATE=false PANTS_TEST_JUNIT_TEST_SHARD=1/2 PANTS_PLUGIN_DEPLOY=true
 
 script:
   # Test a single Java target without scala plugin
   - ENABLE_SCALA_PLUGIN=false ./scripts/run-tests-ci.sh --test-junit-test=com.twitter.intellij.pants.integration.OSSPantsJavaExamplesIntegrationTest
   - travis_wait 105 ./scripts/run-tests-ci.sh
 
 after_success:
-  - scripts/deploy/deploy.sh
+  - if [ "PANTS_PLUGIN_DEPLOY" = "true" -a "$TRAVIS_BRANCH" = "master" ]; then scripts/deploy/deploy.sh; fi
```

Comment:
I have added this `if` statement here, but not sure if it's the right thing. Please correct me if I'm wrong, but it looks like we have had four jobs running in parallel, and the one that was the fastest was uploading artifacts to the JetBrains repo. @tpasternak @wisechengyi @odisseus

Reply:
shouldn't be necessary. For PR builds, there's no creds available for upload.

Reply:
and yes for master builds, the fastest shard will upload, but that hasn't been an issue.

Reply:
the deploy script already has this logic:
```$ cat scripts/deploy/deploy.sh
#!/bin/bash
# Trigger deploy process if travis ci build is on master or on a tag.
if [ "$TRAVIS_BRANCH" == "master" ] || [ ! -z "$TRAVIS_TAG" ]; then
```